compiler/rustc_mir_build/src/build/expr/as_place.rs

```diff
@@ -499,22 +500,23 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 }
                 block.and(place_builder)
             }
-            ExprKind::ValueTypeAscription { source, ref user_ty } => {
+            ExprKind::ValueTypeAscription { source, ref user_ty, user_ty_span } => {
                 let source_expr = &this.thir[source];
                 let temp = unpack!(
                     block = this.as_temp(block, source_expr.temp_lifetime, source, mutability)
                 );
                 if let Some(user_ty) = user_ty {
+                    let ty_source_info = this.source_info(user_ty_span);
                     let annotation_index =
                         this.canonical_user_type_annotations.push(CanonicalUserTypeAnnotation {
-                            span: source_info.span,
+                            span: user_ty_span,
                             user_ty: user_ty.clone(),
                             inferred_ty: expr.ty,
                         });
                     this.cfg.push(
                         block,
                         Statement {
-                            source_info,
+                            source_info: ty_source_info,
```

Comment:
this feels weird, I don't have much context, but are you sure this is correct? shouldn't `source_info` for a statement cover the whole statement?

Reply:
This matches what we do `AscribeUserType` in other places like the type of [let statements](https://github.com/rust-lang/rust/blob/7c7372b6a1b9a8d6b787a3e5a10e4f8325b1fdfc/compiler/rustc_mir_build/src/build/matches/mod.rs#L638) or [patterns](https://github.com/rust-lang/rust/blob/7c7372b6a1b9a8d6b787a3e5a10e4f8325b1fdfc/compiler/rustc_mir_build/src/build/matches/mod.rs#L2559).

Reply:
Okay 👍🏻